mono/mini/jit-icalls.c

```diff
@@ -93,8 +94,28 @@ ldvirtfn_internal (MonoObject *obj, MonoMethod *method, gboolean gshared)
 	}
 
 	/* An rgctx wrapper is added by the trampolines no need to do it here */
+	gboolean need_unbox = m_class_is_valuetype (res->klass) && !m_class_is_valuetype (method->klass);
+	if (need_unbox && !mono_use_interpreter) {
```

Comment:
Is `!mono_use_interpreter` meant to fix something or is it just a left over from the initial PR ?

Reply:
Will investigate.

Reply:
I did some test with or without this interpreter check and added my comment to the initial PR. So this check probably needs to be removed and something needs to be fixed for failing test. I am not sure if we should have a follow-up PR for that issue or have it fixed in this PR as well. That problem is related to fullAOT+interp.

Reply:
The problem doesn't seem to be present any more.